elasticsearch/docker/dockerfiles/AL2.dockerfile

```diff
@@ -0,0 +1,73 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# This dockerfile generates an AmazonLinux-based image containing an OpenDistro for Elasticsearch (ODFE) installation.
+# It assumes that the working directory contains four files: an ODFE tarball (odfe.tgz), log4j2.properties, elasticsearch.yml, docker-entrypoint.sh.
+# Build arguments:
+#   ODFE_VERSION: Required. Used to label the image.
+#   BUILD_DATE: Required. Used to label the image. Should be in the form 'yyyy-mm-ddThh:mm:ssZ', i.e. a date-time from https://tools.ietf.org/html/rfc3339. The timestamp must be in UTC.
+#   UID: Optional. Specify the elasticsearch userid. Defaults to 1000.
+#   GID: Optional. Specify the elasticsearch groupid. Defaults to 1000.
+
+FROM amazonlinux:2 AS build
+
+ARG ODFE_VERSION
+ARG BUILD_DATE
+ARG UID=1000
+ARG GID=1000
+
+RUN yum install -y tar gzip shadow-utils
+
+# Create an elasticsearch user, group, and directory
+RUN groupadd -g $GID elasticsearch && \
+    adduser -u $UID -g $GID -d /usr/share/elasticsearch elasticsearch
+
+RUN mkdir /tmp/elasticsearch
+
+COPY odfe.tgz /tmp/elasticsearch/odfe.tgz
```

Comment:
`odfe.tgz`? Maybe use the actual name of tarball?

Reply:
The build script is responsible for figuring out the actual name of the tarball - I want to move as much logic as possible out of the Dockerfile.

If we have a version-named file, then we would have to pass that as a build argument. It's simpler to just have the build script rename the versioned odfe file to "odfe.tgz" than to have the script obtain the version number, pass it to the Docker build, and have the Dockerfile then reconstruct the filename.

Reply:
That sounds good. Thanks